### PR TITLE
Darwin

### PR DESCRIPTION
**Pull request by kryptonite303**
### Description

Adds support for Mac systems and also allows to set up evergreen config for a specified user using the evergreen_override_user variable.

### Testing

```
make test
```

### CODE REVIEW

**Comment by jasonwalsh** [comment]
The `evergreen_override_user` should be defined as a variable in `defaults/main.yml`.

File: tasks/setup-Darwin.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Copy evergreen.yml
+  template:
+    src: evergreen.yml
+    dest: "~{{ evergreen_user }}/.evergreen.yml"
```
Comment (jasonwalsh) [inline tasks/setup-Darwin.yml]: Does this actually expand? You don't have a test for it in your InSpec tests...

File: defaults/main.yml [hunk 2]
```diff
@@ -1 +1,2 @@
 ---
+evergreen_override_user: null
```
Comment (jasonwalsh) [inline defaults/main.yml]: Shouldn't this be an empty string as the zero value? A null type and a string type are two different data types.

**Comment by kryptonite303** [comment]
The test is already in molecule/default/tests/test_default.rb
I just added the variable `evergreen_override_user` to the playbook to test that the file is actually at `/root/.evergreen.yml`

File: tasks/setup-Darwin.yml (hunk 1 above)
Comment (jasonwalsh) [reply tasks/setup-Darwin.yml]: You resolved this but didn't comment?

**Comment by jasonwalsh** [comment]
`setup-Debian` and `setup-Darwin` are doing the same exact thing. You should create a `main.yml` to consolidate both files.

File: tasks/setup-Darwin.yml (hunk 1 above)
Comment (kryptonite303) [reply tasks/setup-Darwin.yml]: So the test for this is in the `playbook.yml`, `evergreen_user` is defined as `root`, so `~{{ evergreen_user }}` resolves to `/root` and this task will place `evergreen.yml` into `/root`

**Pull request merged**